pennylane/ops/op_math/controlled.py

```diff
@@ -556,8 +556,14 @@ def has_generator(self):
 
     def generator(self):
         sub_gen = self.base.generator()
-        proj_tensor = operation.Tensor(*(qml.Projector([1], wires=w) for w in self.control_wires))
-        return 1.0 * proj_tensor @ sub_gen
+        projectors = (
+            qml.Projector([val], wires=w) for val, w in zip(self.control_values, self.control_wires)
+        )
+
+        if qml.operation.active_new_opmath():
+            return qml.prod(*projectors, sub_gen)
+
+        return 1.0 * operation.Tensor(*projectors) @ sub_gen
```

Comment:
Huh this seems to have introduced an error for some reason, investigating...

Reply:
Ok the purpose of `1.0 * ...` is to convert the `Tensor` to a `Hamiltonian`, because `_generator_prefactor` in `generator.py` does not know what to do with `Tensor` objects. I'm adding the `1.0 * ` term back in, or @timmysilv do you have a suggestion?

Reply:
🤦 sorry about that suggestion. We can add that back in.

Reply:
Yeah I'm reverting the way it was before. It turns out it was written like that for a reason. o(*≥▽≤)ツ┏━┓